gradle/ide.gradle

```diff
@@ -21,6 +24,38 @@ allprojects {
   }
 }
 
+tasks.register('configureIdeCheckstyle') {
+  group = 'ide'
+  description = 'Generated a suitable checkstyle config for IDEs'
+
+  doLast {
+    // Create an IDE-specific checkstyle config by first copying the standard config
+    Files.copy(
+      Paths.get(project.file("buildSrc/src/main/resources/checkstyle.xml").getPath()),
+      Paths.get(project.file("buildSrc/src/main/resources/checkstyle_ide.xml").getPath()),
```

Comment:
generated files should live in the `build` folder in general. any particular reason its not

Reply:
Otherwise we tangle with the up-to-date check for the processResources task here as src/main/resources is considered its input

Reply:
I've moved the generated config to 'build'.